Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h

```diff
@@ -0,0 +1,120 @@
+ * The transition from ideal to real geometry (i.e. loading alignments) is done by TotemRPRealGeometryModule.
+ * 
+ * The <b>translation</b> and <b>rotation</b> parameters are defined by <b>local-to-global</b>
+ * coordinate transform. That is, if r_l is a point in local coordinate system and x_g in global,
+ * then the transform reads:
+ \verbatim
+    x_g = rotation * x_l + translation
+ \endverbatim
+ **/
+
+class DetGeomDesc
+{
+	public:
+		typedef std::vector< const DetGeomDesc*>  ConstContainer;
+		typedef std::vector< DetGeomDesc*>  Container;
+		typedef DDExpandedView::nav_type nav_type;
+		
+		/// a type (not used in the moment, left for the future)
+		typedef unsigned int GeometricEnumType;
+		
+		///Constructors to be used when looping over DDD
+		DetGeomDesc(nav_type navtype, GeometricEnumType dd = 0);
+		DetGeomDesc(DDExpandedView* ev, GeometricEnumType dd = 0);
+		DetGeomDesc(DDFilteredView* fv, GeometricEnumType dd = 0);
+		
+		/// copy constructor and assignment operator
+		DetGeomDesc(const DetGeomDesc &);
+		DetGeomDesc& operator= (const DetGeomDesc &);
+
+		/// destructor
+		virtual ~DetGeomDesc();
+		
+		/// ID stuff
+		void setGeographicalID(DetId id) { _geographicalID = id; }
+		virtual DetId geographicalID() const { return _geographicalID; }
+
+		/// access to the tree structure
+		virtual ConstContainer components() const;
+		virtual Container components();
+		virtual ConstContainer deepComponents() const;				/// returns all the components below
+		virtual std::vector< DDExpandedNode > parents() const		/// retuns the geometrical history
+			{ return _parents; }
+
+		/// components (children) management
+		void setComponents(Container cont)
+			{ _container = cont; }
+		void addComponents(Container cont);
+		void addComponent(DetGeomDesc*);
+		void clearComponents()
+			{ _container.resize(0);} 
+		void deleteComponents(); 									/// deletes just the first daughters
+		void deepDeleteComponents();  								///traverses the treee and deletes all nodes.
+		bool isLeaf() const 
+			{ return (_container.size() == 0); }
+		
+		/// geometry information
+		DDRotationMatrix	rotation() const {return _rot;}
+		DDTranslation		translation() const {return _trans;}
+		DDSolidShape		shape() const  {return _shape;}
+		GeometricEnumType	type() const {return _type;}
+		DDName				name() const {return _ddname;};
+		nav_type			navType() const {return _ddd;}
+		std::vector<double>	params() const {return _params;}
+		virtual int			copyno() const {return _copy;}
+		virtual double		volume() const {return _volume;}
+		virtual double		density() const {return _density;}
+		virtual double		weight() const {return _weight;}
+		virtual std::string	material() const {return _material;}
+
+		/// alignment
+		void ApplyAlignment(const RPAlignmentCorrectionData&);
+		
+	private:
+		Container						_container;
+		nav_type 						_ddd;	
+		DDTranslation 					_trans;
```

Comment:
@grzanka - you may consider Surface::PositionType and Surface::RotationType instead of DD classes. Especially if this class is going to be used in reconstruction.